### DISCUSSION TITLE
How to save or record detail from llama_print_timings.

### DISCUSSION THREAD

**Discussion by SiraHaruethaipree** [abetlen/llama-cpp-python#650] [Q&A]
I have no idea about to collect these timings of operation. I try to define a variable it always show None or empty. 
Thanks.

![image](https://github.com/abetlen/llama-cpp-python/assets/82432680/63482562-e45a-4a88-8f33-c1d250e572ec)

**Comment by ianscrivener** [comment]
These are in  the `stderr` output from the underlying [llama.cpp](https://github.com/ggerganov/llama.cpp) library. I did some code a while ago (in node.js) with these. TL;DR you'll likely have grab the strings from `stderr`and crunch the strings from there.

**Reply by JayKayNJIT** [reply]
This solution worked for me. Code snippet below (provide your own file_path):

from io import StringIO

stdout_orig = sys.stdout
stderr_orig = sys.stderr
sys.stdout = StringIO()
sys.stderr = StringIO()

stdout_content = sys.stdout.getvalue()
stderr_content = sys.stderr.getvalue()

sys.stdout = stdout_orig
sys.stderr = stderr_orig

with open(file_path, "a") as file:
    file.write("Stderr Output:\n")
    file.write(stderr_content + "\n")

**Comment by SiraHaruethaipree** [comment]
After I try to use `stderr` as you suggest. the result still empty in python. I'm not sure as I do is correct or not. 

![image](https://github.com/abetlen/llama-cpp-python/assets/82432680/10f65f5d-fbf4-40d3-aa25-092e0c15465b)
